python/runtime/pai/tensorflow_submitter/train.py

```diff
@@ -125,3 +130,130 @@ def train(datasource,
                            feature_columns_code, num_workers)
         print("Model saved to oss: %s" % oss_model_dir)
     print("Done training")
+
+
+# TODO(typhoonzero): used for codegen/experimental, called by
+# `runtime.pai.entry`.
+def train_step(original_sql,
+               model_image,
+               estimator_string,
+               datasource,
+               select,
+               validation_select,
+               model_params,
+               train_params,
+               feature_column_map,
+               label_column,
+               save,
+               load=None):
+    conn = db.connect_with_data_source(datasource)
+    fc_map_ir, fc_label_ir = infer_feature_columns(conn,
+                                                   select,
+                                                   feature_column_map,
+                                                   label_column,
+                                                   n=1000)
+    fc_map = compile_ir_feature_columns(fc_map_ir, EstimatorType.TENSORFLOW)
+    field_descs = get_ordered_field_descs(fc_map_ir)
+    feature_column_names = [fd.name for fd in field_descs]
+    feature_metas = dict([(fd.name, fd.to_dict()) for fd in field_descs])
+    label_meta = label_column.get_field_desc()[0].to_dict()
+
+    feature_column_names_map = dict()
```

Comment:
Seems that `feature_column_names_map` is not used?

Reply:
Added back saving model meta to OSS, it will use `feature_column_names_map`.